### DISCUSSION TITLE
Tasmota Thermostat internal temperature sensor (NSPanel)

### DISCUSSION THREAD

**Discussion by soyxan** [arendst/Tasmota#17359] [Q&A]
Hi, I want to use the Tasmota Thermostat feature in my NSPanel device, so I build my own firmware with the changes in "user_config_override.h" as per the docs (https://tasmota.github.io/docs/Thermostat/). The only change I made to the "defines" in the docs is to set the ADC sensor instead of the default one (DS18B20):
```
#ifdef THERMOSTAT_SENSOR_NAME
#undef  THERMOSTAT_SENSOR_NAME
#define THERMOSTAT_SENSOR_NAME                "ANALOG" // Name of the local sensor to be used
#endif
```
The name of the sensor (ANALOG) is obtained from the topic "tele/tasmota_FA4C08/SENSOR" which has the following value:
`{"Time":"2022-12-12T09:30:50","ANALOG":{"Temperature1":22.6},"Thermostat0":{"ThermostatModeSet":1,"TempTargetSet":24.50,"CtrDutyCycleRead":66},"TempUnit":"C"}`
The firmware is flashed correctly, and I have issued the command SensorInputSet 1 to use the local sensor instead of MQTT.
The thermostat is working (activates the relay output), but the current temperature is shown as 0ºC on the main page of the web UI:
```
Analog1 Temperature    22.8 °C
Thermostat    Enabled
Set Point Temperature    24.5 °C
Current Temperature    0.0 °C
Gradient Temperature    0.0 °C
Duty cycle    66 %
Cycle time    30 Min
PI Auto tuning    Disabled
```

**Comment by soyxan** [comment]
A workaround I am using at this moment that works is to use the rule ON ANALOG#Temperature1 DO TempMeasuredSet %value% ENDON which sets the right temperature reading, which is working but I was wondering if there should be a better why to do this, like using a define in the user_config_override.h (similar to #define THERMOSTAT_RELAY_NUMBER) or trough some command (similar to the command OutputRelaySet)

Further discussion in #17365

**Reply by barbudor** [reply]
This is indeed the only solution as this have already been addressed including a few weeks back ... If you only have searched
https://github.com/arendst/Tasmota/discussions/17168

No need to open an issue, there will be no change
